python-package/mc2client/core.py

```diff
@@ -556,6 +565,11 @@ def generate_keypair(expiration=10 * 365 * 24 * 60 * 60):
 
     logger.info("Generated private key and outputted to {}".format(private_key_path))
 
+    with open(public_key_path, "wb") as public_key_file:
+        public_key_file.write(crypto.dump_publickey(crypto.FILETYPE_PEM, key))
```

Comment:
This line writes the private key - this should be updated to writing the public key